### PR TITLE
Trev8939/viewshed location

### PR DESCRIPTION
**Pull request by TADraeseke**
Adds a sample demonstrating viewshed with a chosen location point.

https://devtopia.esri.com/runtime/common-samples/issues/459

### CODE REVIEW

File: java/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+        // add 50 meters to surface point for location point
+        mViewshed.setLocation(new Point(surfacePoint.getX(), surfacePoint.getY(), surfacePoint.getZ() + 50));
+
+        // ignore default double touch drag gesture
+        return false;
+      }
+    });
+
+    // inflate UI elements
```
Comment (shellygill) [inline java/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.java]: This code is not inflating views.

File: java/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+  private void handleUiElements() {
+
+    mSceneView.setOnTouchListener(new DefaultSceneViewOnTouchListener(mSceneView) {
+
+      @Override public boolean onDoubleTouchDrag(MotionEvent motionEvent) {
+
+        // convert from screen point to surface point
+        android.graphics.Point screenPoint = new android.graphics.Point(Math.round(motionEvent.getX()),
+            Math.round(motionEvent.getY()));
+        Point surfacePoint = mSceneView.screenToBaseSurface(screenPoint);
```
Comment (shellygill) [inline java/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.java]: Specific reason why this is using a slightly different code path to the equivalent JavaSE sample?

File: java/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+
+        // add 50 meters to surface point for location point
+        mViewshed.setLocation(new Point(surfacePoint.getX(), surfacePoint.getY(), surfacePoint.getZ() + 50));
+
+        // ignore default double touch drag gesture
+        return false;
```
Comment (shellygill) [inline java/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.java]: Should return true if the event has been handled.

**Comment by shellygill** [comment]
@TADraeseke Few small comments for you.

File: java/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+
+  /**
+   * Handles double touch drag for movement of viewshed location point, inflation of UI elements, and listeners for
+   * changes in seek bar progress.
+   */
+  private void handleUiElements() {
+
+    mSceneView.setOnTouchListener(new DefaultSceneViewOnTouchListener(mSceneView) {
+
+      @Override public boolean onDoubleTouchDrag(MotionEvent motionEvent) {
```
[the remark this answers is not in the document]
Comment (TADraeseke) [reply java/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.java]: oops! Thanks!

File: java/viewshed-location/src/main/res/layout/viewshed_seek_bars.xml [hunk 5]
```diff
@@ -0,0 +1,165 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout
+        xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
```
[the remark this answers is not in the document]
Comment (TADraeseke) [reply java/viewshed-location/src/main/res/layout/viewshed_seek_bars.xml]: fixed layout_width on 'Vertical Angle' from 87 dip to wrap_content.

File: java/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.java (hunk 2 above)
Comment (TADraeseke) [reply java/viewshed-location/src/main/java/com/esri/arcgisruntime/sample/viewshedlocation/MainActivity.java]: It makes a noticeable performance hit during the drag (being async), but I suppose .screenToLocationAsync(...) is technically more accurate given this is a viewshed sample.

**Comment by TADraeseke** [comment]
@shellygill thanks, all fixes in.

Only fix worth a possible brief further chat is the use of .screenToLocationAsync(...) (switched to this now to bring in line with java se sample) VS .screenToBaseSurface(...). There is a performance hit with .screenToLocationAsync(...) though it is technically more accurate.

**Pull request merged**